ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -69,6 +74,8 @@
 class ReplicaThread implements Runnable {
 
   private final Map<DataNodeId, List<RemoteReplicaInfo>> replicasToReplicateGroupedByNode;
+  private final Set<PartitionId> replicatedPartitions = new HashSet<>();
```

Comment:
One of them is never modified. The other one is modified only under a lock - it doesn't seem necessary to put them under a lock